src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/ServerCertificateTest.cs

```diff
@@ -0,0 +1,183 @@
+            {
+                HttpResponseMessage response = await client.GetAsync(uri);
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.True(_validationCallbackHistory.WasCalled);
+                
+                ConfirmValidCertificate(HttpTestServers.Host);
+            }
+        }
+
+        [Fact]
+        public async Task UseCallback_CallbackReturnsFailure_ThrowsInnerSecurityFailureException()
+        {
+            const int ERROR_WINHTTP_SECURE_FAILURE = 12175;
+
+            var handler = new WinHttpHandler();
+            handler.ServerCertificateValidationCallback = CustomServerCertificateValidationCallback;
+            using (var client = new HttpClient(handler))
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, HttpTestServers.SecureRemoteGetServer);
+                _validationCallbackHistory.ReturnFailure = true;
+                HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() =>
+                    client.GetAsync(HttpTestServers.SecureRemoteGetServer));
+                var innerEx = (Win32Exception)ex.InnerException;
+                Assert.Equal(ERROR_WINHTTP_SECURE_FAILURE, innerEx.NativeErrorCode);
+            }
+        }
+
+        [Fact]
+        public async Task UseCallback_CallbackThrowsSpecificException_ThrowsInnerSpecificException()
+        {
+            var handler = new WinHttpHandler();
+            handler.ServerCertificateValidationCallback = CustomServerCertificateValidationCallback;
+            using (var client = new HttpClient(handler))
+            {
+                _validationCallbackHistory.ThrowException = true;
+                await Assert.ThrowsAsync<CustomException>(() => client.GetAsync(HttpTestServers.SecureRemoteGetServer));
+            }
+        }
+
+        private void ConfirmValidCertificate(string expectedHostName)
+        {
+                Assert.Equal(SslPolicyErrors.None, _validationCallbackHistory.SslPolicyErrors);
+                Assert.True(_validationCallbackHistory.CertificateChain.Count > 0);
+                _output.WriteLine("Certificate.Subject: {0}", _validationCallbackHistory.Certificate.Subject);
+                _output.WriteLine("Expected HostName: {0}", expectedHostName);
+                Assert.Contains(expectedHostName, _validationCallbackHistory.Certificate.Subject);
+        }
+        
+        private bool CustomServerCertificateValidationCallback(
+            HttpRequestMessage sender,
+            X509Certificate2 certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            _validationCallbackHistory.WasCalled = true;
+            _validationCallbackHistory.Certificate = certificate;
+            foreach (var element in chain.ChainElements)
+            {
+                _validationCallbackHistory.CertificateChain.Add(element.Certificate);
+            }
```

Comment:
Not at EOF but right here, where the comment is. New lines after closing brace (}) if followed by anything else but another closing brace.